urle27f76cced_iconmaster5326_HPPPlus_tgz-pJ8-com_iconmaster_hppplus_HPPPlusJ8/src/com/iconmaster/hppplus/parse/element/block/ElementFunction.java

```diff
@@ -0,0 +1,128 @@
+
+package com.iconmaster.hppplus.parse.element.block;
+
+import com.iconmaster.hppplus.HPPPlus;
+import com.iconmaster.hppplus.Parseable;
+import com.iconmaster.hppplus.SourceRange;
+import com.iconmaster.hppplus.exception.HPPPlusException;
+import com.iconmaster.hppplus.exception.parse.EndExpectedException;
+import com.iconmaster.hppplus.exception.parse.SyntaxException;
+import com.iconmaster.hppplus.parse.Element;
+import com.iconmaster.hppplus.parse.Parser;
+import com.iconmaster.hppplus.parse.element.operator.ElementCall.CallData;
+import com.iconmaster.hppplus.parse.element.ElementVariable;
+import com.iconmaster.hppplus.parse.element.operator.ElementCall;
+import com.iconmaster.hppplus.tokenize.Token;
+import com.iconmaster.hppplus.tokenize.element.TokenWord;
+import java.util.ArrayList;
+
+/**
+ *
+ * @author iconmaster
+ */
+public class ElementFunction implements Element {
+    public class FunctionData {
+        public String name;
+        public ArrayList<Element> block;
+        public ArrayList<Element> args;
+        private FunctionData(String name,ArrayList<Element> block,ArrayList<Element> args) {
+            this.name = name;
+            this.args = args;
+            this.block = block;
+        }
+    }
+    
+    private SourceRange range;
+    private String name;
+    private ArrayList<Element> block;
+    private ArrayList<Element> args;
+    private int blockSize;
+    
+    public ElementFunction() {
+        
+    }
+
+    public ElementFunction(String name,ArrayList<Element> block,ArrayList<Element> args,int blockSize,SourceRange range) {
+        this.name = name;
+        this.block = block;
+        this.args = args;
+        this.blockSize = blockSize;
+        this.range = range;
+    }
+    
+    @Override
+    public Object getParsedContent() {
+        return new FunctionData(name,block,args);
+    }
+
+    @Override
+    public SourceRange getRange() {
+        return range;
+    }
+
+    @Override
+    public boolean isMatch(ArrayList<Parseable> parseList, int at) {
+        if (at+2>=parseList.size()) {return false;}
+        if (parseList.get(at) instanceof TokenWord) {
+            Token e = (Token) parseList.get(at);
+            return "FUNCTION".equalsIgnoreCase((String) e.getParsedContent());
+        }
+        return false;
+    }
+
+    @Override
+    public Element match(ArrayList<Parseable> parseList, int at) throws HPPPlusException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.